std/assembly/string.ts

```diff
@@ -432,8 +432,9 @@ export class String {
     return len;
   }
 
-  toUTF8(): usize {
-    var buf = memory.allocate(<usize>this.lengthUTF8);
+  toUTF8(len: i32 = -1): usize {
+    if (len == -1) len = this.lengthUTF8;
+    var buf = memory.allocate(<usize>len);
```

Comment:
But, won't this write out of bounds below then?

Reply:
Did you mean `MAX_LENGTH`?

Reply:
No, in toUTF8, if you allocate let's say 16 bytes, but the string requires 32 bytes, with your change that'd write out of bounds.

Reply:
Now I see. So wdyt about add additional check for loop condition?

Reply:
Fixed. And add tests